### ISSUE TITLE
CodeFixes do not appear anymore in VS2015 Update2

### ISSUE THREAD

**Issue by bkoelman** [dotnet/roslyn-analyzers#1042]
#### Analyzer package
Microsoft.ApiDesignGuidelines.Analyzers

#### Analyzer
CA1036: OverrideMethodsOnComparableTypesAnalyzer

#### Repro steps

1. Open [Analyzers.sln](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Analyzers.sln) in VS2015 Update2
2. Set `Microsoft.ApiDesignGuidelines.Analyzers.Setup` as startup project, press F5
3. In the second instance (RoslynDev hive), create a new Console Application
4. Edit the .csproj: add `<Features>IOperation</Features>` in the first property group
5. At the end of Program.cs, type:
```csharp
public class Test : IComparable<Test> { }
```
6. CS0535 ("Tester does not implement...") appears. Apply its CodeFix "Implement Interface"
7. Hover over `Tester`. It shows CA1036, but no fix is available. Press `Ctrl+.` only shows options Extract Interface... and Suppress.

#### Expected behavior
The CodeFix for CA1036 is offered.

#### Actual behavior
Compiler-built-in CodeFix is available, but CA1036 is not. I have tried a few others, but none of the CodeFixes from this repo are offered. To test, I changed the message for CA1036 and I see that change, so what I compiled is actually being used. But no breakpoint in FixProvider is ever hit.

For reference, the analyzer is [here](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.ApiDesignGuidelines.Analyzers/Core/OverrideMethodsOnComparableTypes.cs) and the FixProvider is [here](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.ApiDesignGuidelines.Analyzers/Core/OverrideMethodsOnComparableTypes.Fixer.cs).

In an attempt to resolve this, I replaced my system with a freshly-installed OS with VS2015 Update2 from backup. That did not solve the issue.

**Comment by davkean** [comment]
tag @srivatsn @heejaechang

**Comment by bkoelman** [comment]
Today I tried VS15 Preview2. Same issue.

As a side note, to get the rep building I had to replace the included
NuGet.exe with 3.4, which supports MSBuild 15. Then rename the projects
under Dependencies, because the new NuGet considered them circular (xUnit
depends on xUnit depends on xUnit etc..)

**Comment by srivatsn** [comment]
@bkoelman - I'll take a look at this later today and get back to you.

**Comment by bkoelman** [comment]
Nice, thanks!

**Comment by mavasani** [comment]
I am unable to repro this on my box with Update2 + InstallDogfood of latest Roslyn development VSIX. I just tried to install the Microsoft.ApiDesignGuidelines.Analyzers 1.2.0-beta2 NuGet package from NuGet.org (not the steps mentioned by @bkoelman), and it seems to work fine. I'll try the VSIX approach and post an update. 

![codefix_ca1036](https://cloud.githubusercontent.com/assets/10605811/15980605/f06ccbda-2f21-11e6-8d2e-15517f7ab6e8.png)

**Comment by mavasani** [comment]
Problem seems to be in our [VSIX source manifest](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.ApiDesignGuidelines.Analyzers/Setup/source.extension.vsixmanifest#L15). We need to specify the analyzer assemblies as Analyzer asset as well as MefComponent asset. The former will suffice for diagnostics and also for fixes when assembly is installed as NuGet package. The later is needed for fixes when installed via VSIX as we MefImport all the code fixers.

**Comment by mavasani** [comment]
I have a fix for this, so taking this up from @srivatsn

**Comment by bkoelman** [comment]
Thanks for the fix, this works for me now.
